src/mon/FSCommands.cc

```diff
@@ -294,6 +308,16 @@ class FsNewHandler : public FileSystemCommandHandler
 
     ss << "new fs with metadata pool " << metadata << " and data pool " << data;
 
+    // set fs options
+    int ret = 0;
+    for(int i = 1 ; i < (int)fsops_vec.size() ; i+=2) {
+      ret = set_val(mon, fsmap, op, cmdmap, ss, fs_name, fsops_vec[i],
+                    fsops_vec[i+1], true);
+      if(ret < 0) {
+        break;
```

Comment:
So the problem here is you can't easily rollback changes to `pending` or the osdmon's `pending` if one of these `set_val` commands fails. I suggest moving the file system creation above the `osdmon()` mutations, running this `set_val` loop, rollback (i.e. delete) the file system on failure, otherwise continue with the `osdmon()` changes.

Reply:
You're talking about these mutations right?
https://github.com/ceph/ceph/blob/2f8a7159a338024ba8897cf625c8f8221d0dfaee/src/mon/FSCommands.cc#L271-L286

Rest other `osdmon()` are just checks

Reply:
Partial completion of commands is generally bad. How do you signal partial success? Read the stdout? Users won't want to do that because it's messy and error-prone.

Reply:
Makes sense, I'll make the required changes

Reply:
btw it would be safe to create a fs before these `osdmon()` mutations right?

Reply:
Yes.

Reply:
pushed the change, PTAL